src/app/core/auth/auth.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from "@angular/core";
+import { Auth } from "aws-amplify";
+import { CognitoHostedUIIdentityProvider } from "node_modules/@aws-amplify/auth/lib-esm/Auth";
+import { CognitoUserSession } from "amazon-cognito-identity-js";
+import { Observable, from } from "rxjs";
+
+@Injectable({
+  providedIn: "root"
+})
+export class AuthService {
+  constructor() {}
+
+  currentSession(): Observable<CognitoUserSession> {
+    return from(Auth.currentSession());
+  }
+
+  signIn(): Promise<any> {
```

Comment:
Why do we need signIn and signOut methods? Hosted UI would take care of that, isn't it?
And even if we need signIn, why federatedSignIn? Isn't that used for social logins like Google / Fb etc.

Reply:
I can see signOut is used and makes sense atm.

Reply:
SignIn and SignOut are aws-amplify methods I'm calling within our service which: build external hosted url, redirect and store needed keys in localstorage. SignOut clears storage and cookies then redirects to hosted UI and clears session.

Reply:
see auth.initializer.ts and status-bar.component.ts for usage